src/bloqade/analysis/address/analysis.py

```diff
@@ -9,8 +10,11 @@
 
 
 class AddressAnalysis(Forward[Address]):
+    """This analysis pass can be used to track the global addresses of qubits."""
+
     keys = ["qubit.address"]
     lattice = Address
+    next_address: int = field(default=0, init=False)
```

Comment:
you don't need `default` if `init=False`?

```suggestion
    next_address: int = field(init=False)
```